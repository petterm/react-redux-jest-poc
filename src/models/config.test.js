import reducer from './config';

describe('config reducer', () => {
  it('should handle update value', () => {
    const NEW_VALUE = 2;
    const state = reducer({
      options: {
        option1: {
          value: 1,
        },
      },
    }, {
      type: 'CONFIG_UPDATE_VALUE',
      payload: {
        name: 'option1',
        value: NEW_VALUE,
      },
    });
    expect(state).toEqual({
      options: {
        option1: {
          value: NEW_VALUE,
        },
      },
    });
  });
});
