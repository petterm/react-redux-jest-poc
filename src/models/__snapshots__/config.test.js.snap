exports[`config actions handles CONFIG_SAVE_SUCCESS action 1`] = `
Array [
  Object {
    "type": "CONFIG_SAVE_START",
  },
  Object {
    "payload": Object {
      "autosave_interval": 30,
      "name": "new name",
    },
    "type": "CONFIG_SAVE_SUCCESS",
  },
]
`;

exports[`config actions handles CONFIG_UPLOAD_SUCCESS action 1`] = `
Array [
  Object {
    "type": "CONFIG_UPLOAD_START",
  },
  Object {
    "payload": Object {
      "autosave_interval": 30,
      "name": "new name",
    },
    "type": "CONFIG_UPLOAD_SUCCESS",
  },
]
`;

exports[`config reducer handles CONFIG_SAVE_FAIL 1`] = `
Object {
  "optionInfo": Object {
    "autosave_interval": Object {
      "defaultValue": 10,
      "description": "Autosave interval in minutes",
      "title": "Autosave interval",
      "type": "number",
    },
    "name": Object {
      "defaultValue": "Default server",
      "description": "Name of the factorio server",
      "title": "Server name",
      "type": "text",
    },
  },
  "optionValues": Object {},
  "saveFailed": true,
  "saving": false,
  "unsavedChanges": true,
  "uploadFailed": false,
  "uploading": false,
}
`;

exports[`config reducer handles CONFIG_SAVE_START 1`] = `
Object {
  "optionInfo": Object {
    "autosave_interval": Object {
      "defaultValue": 10,
      "description": "Autosave interval in minutes",
      "title": "Autosave interval",
      "type": "number",
    },
    "name": Object {
      "defaultValue": "Default server",
      "description": "Name of the factorio server",
      "title": "Server name",
      "type": "text",
    },
  },
  "optionValues": Object {},
  "saveFailed": false,
  "saving": true,
  "unsavedChanges": false,
  "uploadFailed": false,
  "uploading": false,
}
`;

exports[`config reducer handles CONFIG_SAVE_SUCCESS 1`] = `
Object {
  "optionInfo": Object {
    "autosave_interval": Object {
      "defaultValue": 10,
      "description": "Autosave interval in minutes",
      "title": "Autosave interval",
      "type": "number",
    },
    "name": Object {
      "defaultValue": "Default server",
      "description": "Name of the factorio server",
      "title": "Server name",
      "type": "text",
    },
  },
  "optionValues": Object {
    "autosave_interval": 30,
    "name": "new name",
  },
  "saveFailed": false,
  "saving": false,
  "unsavedChanges": false,
  "uploadFailed": false,
  "uploading": false,
}
`;

exports[`config reducer handles CONFIG_UPDATE_VALUE 1`] = `
Object {
  "optionInfo": Object {
    "autosave_interval": Object {
      "defaultValue": 10,
      "description": "Autosave interval in minutes",
      "title": "Autosave interval",
      "type": "number",
    },
    "name": Object {
      "defaultValue": "Default server",
      "description": "Name of the factorio server",
      "title": "Server name",
      "type": "text",
    },
  },
  "optionValues": Object {
    "name": "newName",
  },
  "saveFailed": false,
  "saving": false,
  "unsavedChanges": true,
  "uploadFailed": false,
  "uploading": false,
}
`;
